packages/wp-checkout/src/components/checkout.js

```diff
@@ -105,46 +105,42 @@ function PaymentMethodsStep( { setStepNumber, isActive, isComplete, availablePay
 	const paymentMethod = usePaymentMethod();
 	const [ , setPaymentMethod ] = usePaymentMethodId();
 
-	// We must always display both the active and inactive version to keep their
-	// data available, using CSS to hide whichever is relevant.
 	return (
 		<React.Fragment>
 			<CheckoutStep
 				isActive={ isActive }
 				isComplete={ isComplete }
 				stepNumber={ 1 }
 				title={ localize( 'Pick a payment method' ) }
-			>
-				<CheckoutPaymentMethods
-					isActive={ isActive }
-					isComplete={ isComplete }
-					availablePaymentMethods={ availablePaymentMethods }
-					onChange={ setPaymentMethod }
-					paymentMethod={ paymentMethod }
-				/>
-
-				<CheckoutNextStepButton
-					value={ localize( 'Continue' ) }
-					onClick={ () => setStepNumber( 2 ) }
-				/>
-			</CheckoutStep>
-			<CheckoutStep
-				summary
-				isActive={ isActive }
-				isComplete={ isComplete }
-				stepNumber={ 1 }
-				title={ localize( 'Payment method' ) }
+				completedTitle={ localize( 'Payment method' ) }
```

Comment:
Since we know here if we're complete or not, and we never need to display both titles at the same time, we can just omit `completedTitle` and make it so that `title={ isComplete ? localize( 'Payment method' ) : localize( 'Pick a payment method' ) }`. 

While I do like the philosophy of providing everything the component needs to do its job and letting it decide how best to do it, in this case since `CheckoutStep` is part of our public API, anything that reduces its usage complexity should help those using it.

Reply:
Yeah, I like this! Great suggestion.